tests/test_modeling_t5.py

```diff
@@ -304,6 +304,16 @@ def create_t5_and_check_t5_generate_with_past_key_value_states(
             output_with_past_cache = model.generate(input_ids[:1], num_beams=2, max_length=5, do_sample=True)
             self.parent.assertTrue(torch.all(output_with_past_cache == output_without_past_cache))
 
+        def create_and_check_t5_model_fp16_forward(
+            self, config, input_ids, decoder_input_ids, attention_mask, decoder_attention_mask, lm_labels,
+        ):
+            model = T5Model(config=config)
+            model.to(torch_device)
+            model.half()
+            model.eval()
+            output = model(input_ids, decoder_input_ids=input_ids, attention_mask=attention_mask)[0]
+            self.parent.assertFalse(torch.isnan(output).any().item())
+
```

Comment:
Cool test, which doesn't look too specific to T5. It would be a good idea to put it in `test_modeling_common` at some point.

Reply:
Agree!